### ISSUE TITLE
Wrong index generation

### ISSUE THREAD

**Issue by diegobernardes** [plentz/lol_dba#37]
I have this tables:

```ruby
create_table :users do |t|
end

create_table:roles do |t|
end

create_table :roles_users, id: false do |t|
    t.references :role
    t.references :user
end
```

when i run the command *rake db:find_indexes* it says to add this index:

```ruby
add_index :roles_users, :left_side_id
```

But, the table does not have this column. 
Did not found anything in the documentation about this.

**Comment by iabdulin** [comment]
same with rails 4.1.6 and lol_dba-1.6.4

**Comment by ArnarTumi** [comment]
Same here, rails 4.0.12 and lol_dba-1.6.4

**Comment by plentz** [comment]
this can be reproduced even in the current tests. I will look at this right now

**Comment by plentz** [comment]
@diegobernardes @panozzaj @MaxPleaner @jplethier @ArnarTumi @iabdulin hey guys, I just released v1.6.6 that should fix this problem. give it a try and let me know if you still have some problem

**Comment by wbotelhos** [comment]
Same here with `rails 4.2.5` and `lol_dba 2.0.3`:

```ruby
class AddMissingIndexes < ActiveRecord::Migration
  def change
    add_index :permissions_roles, :left_side_id
    add_index :plans, :unit_id
  end
end
```

By the way the column `unit_id` was removed but appears here:

The migration that removes it:

```ruby
class RemoveUnitFromPlans < ActiveRecord::Migration
  def change
    remove_column :plans, :unit_id, :reference
  end
end
```

The current unit schema:

```ruby
create_table "units", force: :cascade do |t|
  t.decimal  "entrance_fee", precision: 15, scale: 10, default: 0.0, null: false
  t.string   "name"
  t.integer  "school_id"
  t.datetime "created_at",   precision: 6
  t.datetime "updated_at",   precision: 6
  t.integer  "pay_day",                                default: 1,   null: false
end

add_index "units", ["school_id"], name: "index_units_on_school_id", using: :btree
```

Any idea?

**Comment by plentz** [comment]
@wbotelhos can you create a test case that reproduces the problem? Why I asking this: we have [this](https://github.com/plentz/lol_dba/blob/master/spec/associations_index_spec.rb#L41) that ensures exactly this(or should guarantee).

**Comment by DexterTheDragon** [comment]
Looks like this is a problem in Rails >= 4.2.3. I created PR #74 that updates the Rails 4.2 version it tests against and shows the failure.

**Comment by plentz** [comment]
hey guys! it should be fixed @ lol_dba 2.1 that I've just released. give it a try :)
